frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/cdi-upload-provider.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import axios, { Canceler } from 'axios';
+import {
+  WatchK8sResource,
+  useK8sWatchResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { RouteKind } from '@console/internal/module/k8s';
+import { RouteModel } from '@console/internal/models';
+import { UPLOAD_STATUS } from './consts';
+
+const routeResource: WatchK8sResource = {
+  kind: RouteModel.kind,
+  namespaced: true,
+  isList: true,
+  fieldSelector: 'metadata.name=cdi-uploadproxy',
+};
```

Comment:
Are we just sending the data to the first route with the name `cdi-uploadproxy`? I don't see a namespace specified. What's to stop any user on the platform from creating a route with that name?

Reply:
Will the end user always have authority to list the route?

Reply:
nothing, but on the other hand CDI can be installed on any namespace, so a fixed namespace is not a solution.
I think `cdi-uploadproxy` is a pretty unique name, correct me if i'm wrong.

Reply:
It's a security issue. We can't upload data to any route with that name. A malicious user could create that route intentionally to intercept file uploads.

Reply:
@rawagner thoughts?

Reply:
you could restrict it only to the namespaces HCO deploys it. It means you will have two, one for upstream and one for downstream.

Reply:
it would be nice to detect if the HCO is deployed and select the namespace automatically

Reply:
So there is a cluster resource CDIConfig object (its named config). In the STATUS of that object there is a uploadProxyURL field. The value is calculated based on a route or ingress with the appropriate annotation. On Open Shift, CDI will create a route to the upload proxy and the value in that field will point to the route. That is the only route you should be uploading to. Note this object is readable by any user

If you are on a pure k8s install, you can create an ingress with the right annotation, and the same field will get updated based on name generated by that Ingress.